wis2node/api/__init__.py

```diff
@@ -18,3 +18,193 @@
+    API_BACKEND_PORT, API_BACKEND_USERNAME, API_BACKEND_PASSWORD
+)
+from wis2node.plugin import load_plugin
+from wis2node.topic_hierarchy import TopicHierarchy
+from wis2node.metadata.discovery import DiscoveryMetadata
+from wis2node.util import walk_path, yaml_load, yaml_write
+
+LOGGER = logging.getLogger(__name__)
+
+
+def load_backend(topic_hierarchy: str,
+                 fuzzy: bool = False) -> Tuple[TopicHierarchy, Any]:
+    """
+    Validate topic hierarchy and load data defs
+
+    :param topic_hierarchy: `str` of topic hierarchy path
+    :param fuzzy: `bool` of whether to do fuzzy matching of topic hierarchy
+                  (e.g. "*foo.bar.baz*).
+                  Defaults to `False` (i.e. "foo.bar.baz")
+
+    :returns: tuple of `wis2node.topic_hierarchy.TopicHierarchy and
+              plugin object
+    """
+
+    LOGGER.debug(f'Validating topic hierarchy: {topic_hierarchy}')
+
+    th = TopicHierarchy(topic_hierarchy)
+
+    if not th.is_valid():
+        msg = 'Invalid topic hierarchy'
+        LOGGER.error(msg)
+        raise ValueError(msg)
+    if th.dotpath not in DATADIR_DATA_MAPPINGS['api_backend'].keys():
+        msg = 'Topic hierarchy not in data mappings'
+        LOGGER.error(msg)
+        raise ValueError(msg)
+
+    LOGGER.debug('Loading plugin')
+
+    defs = {
+        'topic_hierarchy': topic_hierarchy,
+        'codepath': DATADIR_DATA_MAPPINGS['api_backend'][th.dotpath],
+        'host': API_BACKEND_HOST,
+        'port': API_BACKEND_PORT,
+        'username': API_BACKEND_USERNAME,
+        'password': API_BACKEND_PASSWORD
+    }
+
+    plugin = load_plugin('api_backend', defs)
+
+    return th, plugin
+
+
+def generate_collection_metadata(mcf: dict) -> dict:
+    """
+    Generate metadata for collection from metadata control file
+
+    :param mcf: `dict` of MCF file
+
+    :returns: `dict` of API collection metadata
+    """
+
+    LOGGER.debug('Parsing discovery metadata')
+
+    dm = DiscoveryMetadata()
+    record = dm.parse_record(mcf)
+    generated = dm.generate(record)
+    parsed = json.loads(generated)
+
+    LOGGER.debug('Creating collection configuration')
+
+    return {
+        'type': 'collection',
+        'title': parsed['properties']['title'],
+        'description': parsed['properties']['description'],
+        'keywords': record['identification']['keywords'],
+        'extents': parsed['properties']['extent'],
+        'links': parsed['links'] + parsed['associations'],
+        'providers': []
+    }
+
+
+def load_collection_items(topic_hierarchy: str) -> None:
+    """
+    Load collection items into collection
+
+    :param topic_hierarchy: `str` of topic hierarchy path
+
+    :returns: None
+    """
+
+    th, backend = load_backend(topic_hierarchy)
+
+    regex = f'.*{th.dirpath}.*.geojson$'
+
+    LOGGER.debug('Loading files matching: {}'.format(regex))
+
+    for file in walk_path(DATADIR_PUBLIC, regex):
```

Comment:
`file` is a Python 2 keyword.  Even though we are using Python 3, rename to something like `file_` for good measure.